src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,180 @@
+            current.prev.next = newNode;
+            newNode.prev = current.prev;
+            current.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T values : list) {
+            add(values);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> elementByIndex = takeElementByIndex(index);
+        return (elementByIndex == null) ? null : elementByIndex.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> elementByIndex = takeElementByIndex(index);
+        if (elementByIndex == null) {
+            return null;
+        }
+        T valueToReturn = elementByIndex.item;
+        elementByIndex.item = value;
+        return valueToReturn;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> elementByIndex = takeElementByIndex(index);
+        return unLink(elementByIndex).item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        for (int index = 0; index < size; index++) {
+            if (current == null) {
+                return false;
+            } else if (Objects.equals(object, current.item)) {
+                unLink(current);
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return (head == null);
+    }
+
+    private class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T item) {
+            this.item = item;
+        }
+
+        public Node(Node<T> next, T item, Node<T> prev) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private Node<T> takeElementByIndex(int index) {
+        Node<T> current;
+        if (index == 0) {
+            return head;
+        } else if (index == size - 1) {
+            return tail;
+        } else {
+            if (index <= (size >> 1)) {
+                current = head;
+                for (int i = 0; i < index; i++) {
+                    if (current == null) {
+                        return null;
+                    } else {
+                        current = current.next;
+                    }
+                }
+            } else {
+                current = tail;
+                for (int i = size - 1; i > index;i--) {
+                    if (current == null) {
+                        return null;
+                    } else {
+                        current = current.prev;
+                    }
+                }
+            }
+        }
```

Comment:
don't complicate if else construction
If you use return inside if, after that you don't need to use else or else if 
You can create another if check